### PR TITLE
Move srand out of weak entropy poll loop

### PR DESCRIPTION
**Pull request by kuggenhoffen**

### CODE REVIEW

**Comment by kuggenhoffen** [comment]
@simosillankorva @korjaa @anttiylitokola @TeroJaasko

**Review by korjaa — APPROVED** [review]
Looks good to me.

File: source/m2mconnectionsecuritypimpl.cpp [hunk 1]
```diff
@@ -393,8 +393,8 @@ int entropy_poll( void *, unsigned char *output, size_t len,
     } else {
         rdm = time(NULL);
     }
+    srand(rdm);
```
Comment (JanneKiiskila) [inline source/m2mconnectionsecuritypimpl.cpp]: Should this use a PAL random number API instead?

**Review by JanneKiiskila — CHANGES_REQUESTED** [review]

File: source/m2mconnectionsecuritypimpl.cpp (hunk 1 above)
Comment (simosillankorva) [inline source/m2mconnectionsecuritypimpl.cpp]: This code will be removed anyway when we merge the PAL TLS version https://github.com/ARMmbed/mbed-client-mbed-tls/pull/63.

**Comment by yogpan01** [comment]
Not valid anymore.

**Pull request closed without merging**